x-pack/plugins/ml/public/embeddables/job_creation/map/map_vis_layer_selection_flyout/layer/compatible_layer.tsx

```diff
@@ -0,0 +1,201 @@
+import {
+  EuiAccordion,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiIcon,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { MapEmbeddable } from '@kbn/maps-plugin/public';
+import {
+  type LayerResult,
+  QuickGeoJobCreator,
+  redirectToGeoJobWizard,
+} from '../../../../../application/jobs/new_job/job_from_map';
+import { useMlFromLensKibanaContext } from '../../../lens/context';
+import { JobDetails, CreateADJobParams } from '../../../common/job_details';
+
+interface DropDownLabel {
+  label: string;
+  field: string;
+}
+
+interface Props {
+  embeddable: MapEmbeddable;
+  layer: LayerResult;
+  layerIndex: number;
+}
+
+export const CompatibleLayer: FC<Props> = ({ embeddable, layer, layerIndex }) => {
+  const [selectedSplitField, setSelectedSplitField] = useState<string | null>(null);
+  const [createError, setCreateError] = useState<{ text: string; errorText: string } | undefined>();
+
+  const {
+    services: {
+      data,
+      share,
+      uiSettings,
+      mlServices: { mlApiServices },
+    },
+  } = useMlFromLensKibanaContext();
+
+  const quickJobCreator = useMemo(
+    () =>
+      new QuickGeoJobCreator(uiSettings, data.query.timefilter.timefilter, share, mlApiServices),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [data, uiSettings]
+  );
+
+  const createGeoJobInWizard = useCallback(() => {
+    redirectToGeoJobWizard(
+      embeddable,
+      layer.dataView!.id!,
+      layer.geoField,
+      selectedSplitField,
+      share
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [layer?.dataView?.id, embeddable, selectedSplitField]);
+
+  const createGeoJob = useCallback(
+    async ({
+      jobId,
+      bucketSpan,
+      embeddable: mapEmbeddable,
+      startJob,
+      runInRealTime,
+    }: CreateADJobParams) => {
+      try {
+        const result = await quickJobCreator.createAndSaveGeoJob({
+          jobId,
+          bucketSpan,
+          embeddable: mapEmbeddable as MapEmbeddable,
+          startJob,
+          runInRealTime,
+          sourceDataView: layer.dataView,
+          geoField: layer.geoField,
+          splitField: selectedSplitField,
+        });
+
+        return result;
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error(error);
+        setCreateError({
+          text: i18n.translate('xpack.ml.embeddables.geoJobFlyout.jobCreationError', {
+            defaultMessage: 'Job could not be created.',
+          }),
+          errorText: error.message ?? error,
+        });
+
+        return {
+          jobCreated: { success: false },
+          datafeedCreated: { success: false },
+          jobOpened: { success: false },
+          datafeedStarted: { success: false },
+        };
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
are these eslint ignores definitely needed?

Reply:
Removed some unnecessary ones but kept the ones where the dependency would be an incoming prop
In 8da1fadcaca00341a6a40c9abda84d799ceddd95